server/channels/app/import_functions.go

```diff
@@ -939,32 +939,42 @@ func (a *App) importProfileImage(rctx request.CTX, userID string, data *imports.
 		var f io.ReadCloser
 		f, err = data.ProfileImageData.Open()
 		if err != nil {
-			rctx.Logger().Warn("Unable to open the profile image data.", mlog.Err(err))
-		} else {
-			limitedReader := io.LimitReader(f, *a.Config().FileSettings.MaxFileSize)
-			var b []byte
-			b, err = io.ReadAll(limitedReader)
-			if err != nil {
-				rctx.Logger().Warn("Unable to read all bytes from profile picture.", mlog.Err(err))
-			} else {
-				file = bytes.NewReader(b)
+			return model.NewAppError("importProfileImage", "app.import.profile_image.open.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
```

Comment:
Can we also pass `data.ProfileImageData.Name` as an additional context? This is regards to our convo around logging guidelines. "Failed to open image file" in the log message invariably gives rise to the question of "which image file?". 

Same applies to other places where we are reading/closing the file handle as well. We need to add the file name as context to all of those.

Reply:
Sure thing, I add the file name to ` "app.import.profile_image.open.app_error"`.

In general, I find it way too painful to add context to `model.AppError`. I wonder if we should migrate files like this one way from using `model.AppError` and use a plain `error` instead. The only place where the custom error type is import is https://github.com/mattermost/mattermost/blob/013baa2718b02458ba70f7d1c63bf83c3ca80f14/server/channels/app/import.go#L31-L42, and we can easily substitute that using `errors.Is` and a global error.